modules/apps/message-boards/message-boards-web/src/main/resources/META-INF/resources/message_boards/js/MBPortlet.es.js

```diff
@@ -55,16 +55,30 @@ class MBPortlet {
 			`${this._namespace}workflowAction`
 		);
 
+		this._events = [];
 		this._attachEvents();
 	}
 
+	dispose() {
+		this._events.forEach(({target, event, listener}) =>
+			target.removeEventListener(event, listener)
+		);
+
+		this._events = [];
+	}
+
+	_addEventListener(target, event, fn) {
+		target.addEventListener(event, fn);
+		this._events.push({target, event, fn});
+	}
+
 	_attachEvents() {
 		const publishButton = this.rootNode.querySelector(
```

Comment:
Not blocking, but one option to make this better could be to make this return the `events` array so we don't have to weirdly mutate `this._events` and push in the `_addEventListener`. 

This is weirdly looking because `addEventListener` does not return a handle, so even then you'd either need some sort of `_addEventListener` or just repeat that which does not seem so terrible in this case.

Just some food for thought here... @wincent 

/cc @bryceosterhaus, this is a different case than the one you're working on for the `delegate` case... I don't think we want to necessarily create a wrapper for `addEventListener`, but maybe having a clearer attach/detach pattern for this would be helpful... just in case you have some thoughts on this context

Reply:
At previous job we had a `SubscriptionHandler` class for these. You could shove `dispose`-able handles in there then release them all at once at come appropriate point in the lifecycle. Worked well, but only because we used the pattern consistently pretty much every where. If you just do it in one spot, the abstraction probably doesn't pull its weight.

```js
constructor() {
   this._subscriptions = new SubscriptionHandler();
}

dispose() {
   this._subscriptions.release();
}
```

And you'd add `dispose`-ables with (something like):

```js
this._subscriptions.add(addEventListener(target, event, fn));
```

Note that `addEventListener` there can be anything that returns an object with a `dispose()` method. We'd use it for things like event listeners, web sockets etc.

Not saying we need this kind of abstraction here though, because we might not even see this kind of thing come up very often now (with more and more code in React nowadays, people are more likely to be able to use `useEventListener()` or React's own event handling).

Reply:
> Not saying we need this kind of abstraction here though, because we might not even see this kind of thing come up very often now (with more and more code in React nowadays, people are more likely to be able to use useEventListener() or React's own event handling).

Yeah, that was my next thought... not saying we want to... but how horrible would it be to do non-rendering react components just for the side effects? 😂 

So, here, rather than instantiating a Class, we could render a function component that rendered nothing but used `useEventListener` or others in a `useEffect` and handle these things always in a similar way.

It just occurred to me that if we treated these side effects in a more similar way as we treat other components we'd have less variants to worry about.

Just a thought... 🤔